### PR TITLE
#2409 Update bank verification API

### PR DESCRIPTION
**Pull request by desfero**

### CODE REVIEW

File: app/modules/kyc/selectors.ts [hunk 1]
```diff
@@ -122,5 +122,19 @@ export const selectIsAccountFrozen = createSelector(selectClaims, claims => {
...
+  }
+
+  return {
+    hasBankAccount: false,
+  };
+};
+
+export const selectIsBankAccountLoading = (state: IAppState): boolean => {
+  return state.kyc.bankAccount === undefined;
+};
```
Comment (AlexeyTuboltsev) [inline app/modules/kyc/selectors.ts]: Are you using state.kyc.bankAccount = undefined as loading state?
I suggest to add a normal loading state then, and some error state as well

File: app/lib/api/KycApi.ts [hunk 2]
```diff
@@ -278,14 +277,10 @@ export class KycApi {
     const response = await this.httpClient.get<IKycRequestState>({
       baseUrl: BASE_PATH,
       url: BANK_ACCOUNT_PATH,
-      responseSchema: KycBankAccountSchema,
-      allowedStatusCodes: [404],
+      // TODO test why `optional` is not working
+      // responseSchema: KycBankAccountSchema,
```
[the remark this answers is not in the document]
Comment (desfero) [reply app/lib/api/KycApi.ts]: there is some bug in the code when `optional` is not working for `responseSchema` will investigate this later

File: app/modules/kyc/selectors.ts [hunk 3]
```diff
@@ -122,5 +122,19 @@ export const selectIsAccountFrozen = createSelector(selectClaims, claims => {
 export const selectIsUserVerifiedOnBlockchain = (state: IAppState) =>
   selectIsClaimsVerified(state) && !selectIsAccountFrozen(state);
 
-export const selectBankAccount = (state: IAppState): DeepReadonly<TBankAccount> | undefined =>
-  state.kyc.bankAccount;
+export const selectBankAccount = (state: IAppState): DeepReadonly<TBankAccount> | undefined => {
+  if (state.kyc.bankAccount && state.kyc.bankAccount.verifiedUserAccount) {
```
[the remark this answers is not in the document]
Comment (desfero) [reply app/modules/kyc/selectors.ts]: fixed

File: app/modules/kyc/sagas.ts [hunk 4]
```diff
@@ -612,11 +619,9 @@ function* loadBankAccountDetails({
       );
     }
```
Comment (AlexeyTuboltsev) [inline app/modules/kyc/sagas.ts]: lines 607-614-620 look funny

**Pull request merged**